clap_derive/src/derives/from_argmatches.rs

```diff
@@ -239,3 +249,171 @@ pub fn gen_constructor(
         #( #fields ),*
     }}
 }
+
+// TODO: update_from_subcommand is missing
+pub fn gen_updater(
+    fields: &Punctuated<Field, Token![,]>,
+    parent_attribute: &Attrs,
+) -> TokenStream {
+    let fields = fields.iter().map(|field| {
+        let attrs = Attrs::from_field(
+            field,
+            parent_attribute.casing(),
+            parent_attribute.env_casing(),
+        );
+        let field_name = field.ident.as_ref().unwrap();
+        let kind = attrs.kind();
+        match &*attrs.kind() {
+            Kind::Subcommand(ty) => {
+                let subcmd_type = match (**ty, sub_type(&field.ty)) {
+                    (Ty::Option, Some(sub_type)) => sub_type,
+                    _ => &field.ty,
+                };
+
+                let val = match **ty {
+                    Ty::Option => quote_spanned!( ty.span()=> Some(val) ),
+                    _ => quote_spanned!( ty.span()=> val ),
+                };
+
+                quote_spanned! { kind.span()=>
+                    if let Some(val) = {
+                        let (name, subcmd) = matches.subcommand();
+                        <#subcmd_type as ::clap::Subcommand>::from_subcommand(
+                            name,
+                            subcmd
+                        )
+                    } {
+                        self. #field_name = #val;
+                    }
+                }
+            }
+
+            Kind::Flatten => quote_spanned! { kind.span()=>
+                ::clap::FromArgMatches::update_from_arg_matches(&mut self. #field_name, matches);
```

Comment:
As a general rule, try to avoid spaces between `self.` and `#field_name` because the space makes them look as two different arguments.
```suggestion
                ::clap::FromArgMatches::update_from_arg_matches(&mut self.#field_name, matches);
```